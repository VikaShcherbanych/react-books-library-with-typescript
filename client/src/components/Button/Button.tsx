import React from "react";
import s from "./Button.module.css";

interface IButtonProps {
  onClick: () => void;
  text: string;
}

const Button: React.FC<IButtonProps> = ({ onClick, text }) => {
  return (
    <button onClick={onClick} className={s.button} type="button">
      {text}
    </button>
  );
};

export default Button;
