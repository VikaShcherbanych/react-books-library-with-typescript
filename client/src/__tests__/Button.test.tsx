import React, { Component } from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from "../components/Button/Button";

configure({adapter: new Adapter()});

type IProps = { onClick: () => void, text: string };

const onClick = jest.fn();
jest.mock("../components/Button/Button.module.css", () => ({}));

describe("Button component", () => {
    let sut: ShallowWrapper<any, Readonly<{}>, Component<{}, {}, any>>;
    let props: IProps;

    beforeEach(() => {
        props = {
                onClick,
                text: "Load more"
        };
        sut = shallow(<Button {...props} />);
    })

    test('should be defined', () => {
        expect(Button).toBeDefined();
    });
    test("renders and matches the snapshot", () => {
        expect(toJson(sut)).toMatchSnapshot();
    });
    test("should call onClick when button is clicked", () => {
        sut.find("button").simulate("click");
        expect(onClick).toHaveBeenCalled();
    });
    test('should render text', () => {
        expect(sut.find("button").text()).toEqual(props.text);
    });

    afterEach(() => {
        jest.clearAllMocks();
      });
});