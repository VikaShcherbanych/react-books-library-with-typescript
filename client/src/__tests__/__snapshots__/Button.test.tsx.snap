// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders and matches the snapshot 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
>
  Load more
</button>
`;
